stdlib/src/utils/static_tuple.mojo

```diff
@@ -254,3 +314,139 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         var base_ptr = Pointer[Self.type].address_of(self.array).address
         var ptr = __mlir_op.`pop.array.gep`(base_ptr, Int(0).value)
         return Pointer(ptr)
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct Array[ElementType: CollectionElement, size: Int](Sized):
+    """A statically sized tuple type which contains elements of homogeneous types.
+
+    Parameters:
+        ElementType: The type of the elements in the tuple.
+        size: The size of the tuple.
+    """
+
+    alias type = __mlir_type[
+        `!pop.array<`, size.value, `, `, Self.ElementType, `>`
+    ]
+    var array: Self.type
+    """The underlying storage for the static tuple."""
+
+    @always_inline
+    fn __init__(inout self):
+        """Constructs an empty (undefined) tuple."""
+        _static_tuple_construction_checks[size]()
+        self.array = __mlir_op.`kgen.undef`[_type = Self.type]()
+
+    @always_inline
+    fn __init__(inout self, *elems: Self.ElementType):
+        """Constructs a static tuple given a set of arguments.
+
+        Args:
+            elems: The element types.
+        """
+        _static_tuple_construction_checks[size]()
+        self = Self()
+        var i = 0
+        for e in elems:
+            self[i] = e[]
+            i += 1
+
+    @always_inline
+    fn __init__(inout self, values: VariadicList[Self.ElementType]):
```

Comment:
**Suggestion** Can we say `*values: Self.ElementType` please?  In general, APIs shouldn't spell out the direct type of `VariadicList` or `VariadicListMem` (this is a sharp edge); instead, they should use the language spelling of `*` to construct the right library type backing the variadic.  In this case, this is actually wrong since you want `VariadicListMem`.  Otherwise, the current code will actually never run destructors for the argument variadic pack.  See how `VariadicListMem` defines `__del__` for example but `VariadicList` can't (since it only works on `AnyRegType`).  This is a QoI thing in the compiler for even *accepting* this code, since you're passing `AnyType` to a thing accepting `AnyRegType`.  This should be a compiler error in a nice world.